### PR TITLE
feat(ci): Cleanup and adding ci

### PR DESCRIPTION
**Pull request by ecktom**

### CODE REVIEW

File: .buildkite/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+- command:
+  - "make lint"
+  label: ":llama: Lint"
+  agents:
+  - "queue=default"
+- command:
+  - "make test"
+  label: ":llama: Test"
+  agents:
+  - "queue=default"
```
Comment (jhedev) [inline .buildkite/pipeline.yaml]: why not build?

Comment (ecktom) [reply .buildkite/pipeline.yaml]: Skipped it as we are not using the bin anyways

Comment (jhedev) [reply .buildkite/pipeline.yaml]: but we ensure that we can build it...

Comment (ecktom) [reply .buildkite/pipeline.yaml]: Added the build step and also some more tests ;)

File: main.go [hunk 2]
```diff
@@ -21,22 +21,43 @@ func (i *executionsArray) Set(value string) error {
...
+
+	os.Exit(exitcode)
+}
+
+func initFlags() {
 	flag.Var(&executions, "execute", "eg. -execute='/usr/local/bin/mybin test123 -o -a' -execute='ls -l'")
 	flag.BoolVar(&stopOnFailure, "stop-on-failure", false, "If true, multiple execute steps get combined with && otherwise ||")
+}
+
+func parseExecuions() (executionsArray, error) {
```
Comment (jhedev) [inline main.go]: parseExecuions -> parseExecutions?

File: main.go [hunk 3]
```diff
@@ -21,22 +21,43 @@ func (i *executionsArray) Set(value string) error {
...
 
-	exitcode := runAllCommands(executions, stopOnFailure)
+	return executions, nil
+}
 
-	os.Exit(exitcode)
+func parseStopOnFailure() bool {
+	flag.Parse()
+
+	return stopOnFailure
```
Comment (jhedev) [inline main.go]: not sure if this method is needed. `flag.Parse` is already executed in `parseExecuions` so stopOnFailure should already be set

**Review by jhedev — COMMENTED** [review]
few thoughts

**Review by jhedev — APPROVED** [review]
endnice.

**Pull request merged**